### PR TITLE
Parameterise the kind/push-images Makefile target

### PR DESCRIPTION
**Pull request by smlx**
# Checklist

- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [x] PR title is ready for changelog and subsystem label(s) applied

This makes it easier to push specific images into an existing local `kind` cluster.

# Closing issues

n/a

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1153,18 +1151,12 @@ kind/port-forwards:
...
 kind/retest:
-		export KUBECONFIG="$$(pwd)/kubeconfig.kind.$(CI_BUILD_TAG)" && \
-		export IMAGE_REGISTRY="registry.$$(./local-dev/kubectl get nodes -o jsonpath='{.items[0].status.addresses[0].address}').nip.io:32080/library" \
-		&& docker login -u admin -p Harbor12345 $$IMAGE_REGISTRY \
-		&& for image in $(KIND_TESTS); do \
-			docker tag $(CI_BUILD_TAG)/$$image $$IMAGE_REGISTRY/$$image:$(SAFE_BRANCH_NAME) \
-			&& docker push $$IMAGE_REGISTRY/$$image:$(SAFE_BRANCH_NAME); \
-		done \
+	export KUBECONFIG="$$(pwd)/kubeconfig.kind.$(CI_BUILD_TAG)" \
+		&& export IMAGE_REGISTRY="registry.$$(./local-dev/kubectl get nodes -o jsonpath='{.items[0].status.addresses[0].address}').nip.io:32080/library" \
```
Comment (tobybellwood) [inline Makefile:1159]: we need to keep a push-images routine in here to push the TEST images - that's mostly when I've been using kind/retest - to iterate ansible playbooks etc.

**Comment by tobybellwood** [comment]
I agree with this, but propose that we categorize the images better, and just make sure to push only the needed images at any stage
```
KIND_SERVICES = api api-db api-redis auth-server broker controllerhandler docker-host drush-alias keycloak keycloak-db webhook-handler webhooks2tasks kubectl-build-deploy-dind ui
KIND_ADDITIONAL_SERVICES = logs2email logs2microsoftteams logs2rocketchat logs2slack logs2webhook
KIND_TEST_SERVICES = local-api-data-watcher-pusher local-git tests
```

kind/test = KIND_SERVICES & KIND_TEST_SERVICES
kind/local-dev-patch = KIND_ADDITIONAL_SERVICES
kind/retest = KIND_TEST_SERVICES

File: Makefile (hunk 1 above)
Comment (smlx) [reply Makefile:1159]: So yes we can do this but what I was going for was a workflow like this for iterating `tests`:

```
rm build/tests
make -j8 build/tests SKIP_SCAN=true
make kind/push-images IMAGES=tests
make kind/retest TESTS='[features-api-variables]'
```

The advantage is that if you are iteratively working on something other than the `tests` image (e.g. `kubectl-build-deploy-dind`) and don't need to push the `tests` image all the time, then you can selectively rebuild and push only the image that you are actually interested in.

Does that make sense?

Comment (tobybellwood) [reply Makefile:1159]: yes, that does (I just read the docs update too...  Happy for this!

**Pull request merged**